bigbang/ingress/listserv.py

```diff
@@ -952,45 +660,8 @@ def get_line_numbers_of_header_starts(
             line_nr for line_nr, line in enumerate(content) if "=" * 73 in line
         ]
 
-    def to_dict(self, include_body: bool = True) -> Dict[str, List[str]]:
-        """
-        Parameters
-        ----------
-        include_body : A boolean that indicates whether the message body should
-            be included or not.
-
-        Returns
-        -------
-        A Dictionary with the first key layer being the header field names and
-        the "body" key. Each value field is a list containing the respective
-        header field contents arranged by the order as they were scraped from
-        the web. This format makes the conversion to a pandas.DataFrame easier.
-        """
-        return ListservListIO.to_dict(self.messages, include_body)
-
-    def to_pandas_dataframe(self, include_body: bool = True) -> pd.DataFrame:
-        """
-        Parameters
-        ----------
-        include_body : A boolean that indicates whether the message body should
-            be included or not.
-
-        Returns
-        -------
-        Converts the mailing list into a pandas.DataFrame object in which each
-        row represents an Email.
-        """
-        return ListservListIO.to_pandas_dataframe(self.messages, include_body)
-
-    def to_mbox(self, dir_out: str, filename: Optional[str] = None):
-        """Safe mailing list to .mbox files."""
-        if filename is None:
-            ListservListIO.to_mbox(self.messages, dir_out, self.name)
-        else:
-            ListservListIO.to_mbox(self.messages, dir_out, filename)
 
-
-class ListservArchive(object):
+class ListservArchive(AbstractArchive):
```

Comment:
I see: "An archive is a list of ListservList elements."

I think this could be made clearer in documentation throughout the library.

Does an Archive, generally speaking, keep track of which list each message is associated with? That would be useful.

Reply:
Here an extended explanation for why I distinguished between mailing lists (currently `AbstractList, W3CList, ListservList`) and archives (currently `AbstractArchive, W3CArchive, ListservArchive`).

**Mailing Archive**:
I defined *archive* in the code as the collection of all Emails -- across time and topics -- that an SDO made publicly accessible. In other words, I used archive to mean the domain name, which for W3C is [@w3.org ](https://lists.w3.org/Archives/Public/)and for 3GPP is [@LIST.ETSI.ORG](https://list.etsi.org/scripts/wa.exe?). The mailing archive is a Python list of *mailing lists* (see the `def __init__(lists: List[AbstractList]):`
), and a mailing list in turn is a Python list of Emails (see the `def __init__(msgs: List[mboxMessage]):`). Thus each Email is associated to a mailing list.
An alternative for the word *archive* could be *library*.

**Mailing list**:
SDOs, such as W3C and 3GPP, group messages into *lists* that focus on specific topics. Examples are [public-abcg@w3.org](https://lists.w3.org/Archives/Public/public-abcg/) and [3GPP_TSG_CT_WG1_E_IMS@LIST.ETSI.ORG](https://list.etsi.org/scripts/wa.exe?A0=3GPP_TSG_CT_WG1_E_IMS).
These mailing lists are accessible from the mailing archive webpage (linked to above in the **Mailing Archive** section) in different ways for different SDOs (as the html code is different) and thus motivates to distinguish between a mailing list and a mailing archive (or what ever word described the latter best).

Reply:
I added improved docstrings to the classes of: `bigbang/ingress/abstract.py, bigbang/ingress/listserv.py, bigbang/ingress/w3c.py`

Reply:
Ok. Thanks for this explanation. I see how it is internally consistent and makes sense.

I believe your use of mailing list is entirely standard.

I have a remaining concern, which is that all the earlier mailing list handling code, built around Mailman and still used for IETF data, uses the term `Archive` to mean something else. So what you've done creates some confusion.

I gather that what you've written is intended for bulk analysis of an SDO's entire library (to use a neutral term for now).

BigBang is used for other use cases. For example, many Mailman instances are not managed by SDOs. The user may not be interested in all the lists associated with the mailing list host.

In the other BigBang code, 'Archive' refers to a locally stored representation of a collection of email, which might include email from multiple lists. It can in principle include some preprocessing. It might not be a good name for it.

But I think maybe a better name for what you are calling an Archive might be an email Host or Domain -- something that is more semiotically linked with the unit of analysis, the SDO's relationship to those lists. 'Archive' might be too generic for this purpose.

Reply:
Thanks for the clarification. Bigbang is just getting better!

I have made the following renaming:
- AbstractList --> AbstractMailingList
- AbstractArchive --> AbstractMailingListDomain
and the same hold for Listserv and W3C.